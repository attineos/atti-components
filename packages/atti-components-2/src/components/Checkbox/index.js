import React from 'react'
import PropTypes from 'prop-types'

import ContainerCheckbox from './styles/ContainerCheckbox'
import FakeCheckbox from './styles/FakeCheckbox'
import HiddenInputCheckbox from './styles/HiddenInputCheckbox'
import LabelCheckbox from './styles/LabelCheckbox'

/**
 * A checkbox button.
 *
 * @version 0.1
 * @author Joseph Lefevre
 */
class Checkbox extends React.Component {
  state = {
    checked: this.props.checked,
  }

  changeState = () => {
    this.setState(prevState => ({
      checked: !prevState.checked,
    }))
  }

  defaultProps = {
    checked: false,
    label: '',
    onChange: null,
    style: {},
  }

  propTypes = {
    /**
     * Whether or not the checkbox is checked.
     */
    checked: PropTypes.bool,
    /**
     * The id of the current element.
     */
    id: PropTypes.string.isRequired,
    /**
     * Content of the label of the checkbox.
     */
    label: PropTypes.string,
    /**
     * The name attribute is used to reference form data after a form is submitted.
     * Numerous checkbox with the same name value will be in the same group.
     */
    name: PropTypes.string.isRequired,
    /**
     * Gets called when the value of the checkbox changes.
     *
     * @param {SyntheticEvent} event The react `SyntheticEvent`
     */
    onChange: PropTypes.func,
    /**
     * Style inline which is given to modify the container
     */
    style: PropTypes.object,
    /**
     * The value attribute of the checkbox. This attribute has meaning when submitting a form.
     */
    value: PropTypes.string.isRequired,
  }

  render() {
    const { id, label, name, onChange, style, value } = this.props
    return (
      <ContainerCheckbox style={style} onClick={this.changeState}>
        <HiddenInputCheckbox
          id={id}
          checked={this.state.checked}
          name={name}
          onChange={onChange}
          type="checkbox"
          value={value}
        />
        <FakeCheckbox />
        {label && <LabelCheckbox htmlFor={id}>{label}</LabelCheckbox>}
      </ContainerCheckbox>
    )
  }
}

export default Checkbox
