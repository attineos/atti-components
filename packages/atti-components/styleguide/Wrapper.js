import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
