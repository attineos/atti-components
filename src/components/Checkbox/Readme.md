Checkbox Button :

```js
<div>
  <Checkbox
    id="checkboxTest"
    name="checkboxTestGroup"
    label="My checkbox button"
    style={{ marginBottom: '8px' }}
    value="ImChecked"
  />
  <Checkbox
    id="checkboxTest2"
    name="checkboxTestGroup"
    label="My second checkbox button"
    value="ImCheckedToo"
  />
</div>
```
