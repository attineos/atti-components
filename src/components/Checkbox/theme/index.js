export default ({ colors, sizes, spaces }) => ({
  checkboxBox: {
    colors: {
      background: colors.secondary,
      backgroundHover: colors.success,
      text: colors.primary,
    },
    sizes: {
      heights: sizes.heights.h6,
      widths: sizes.widths.w6,
      widthCheckSign: sizes.widths.w3,
    },
    spaces: {
      marginRight: spaces.s5,
    },
  },
})
