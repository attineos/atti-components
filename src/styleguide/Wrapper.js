import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
