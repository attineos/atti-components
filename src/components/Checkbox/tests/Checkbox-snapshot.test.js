/* This file was generated automatically, please do not write in it. */
import React from 'react';
import Checkbox from '..';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('Checkbox renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <Checkbox checked={true} id="Atticomponents" label="Atticomponents" name="Atticomponents" onChange={mock} style={{}} value="Atticomponents" ></Checkbox>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    