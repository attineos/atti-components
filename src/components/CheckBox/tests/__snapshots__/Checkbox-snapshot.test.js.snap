// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<span
  className="atti-box"
>
  <span
    className="Atticomponents Checkbox-jv3phg-0 ianYSK CheckBoxBoxContainer-sc-1389qor-0 gsLTLX"
  >
    <input
      checked={true}
      className="HiddenInputCheckBox-sc-1w4cuhl-0 bZqbzq"
      id="Atticomponents"
      name="Atticomponents"
      onChange={[Function]}
      type="checkbox"
      value="Atticomponents"
    />
    <span
      className="CheckBoxBox-sc-6daqt7-0 liLBsl"
    />
  </span>
</span>
`;
